### DISCUSSION TITLE
Features for time series classification

### DISCUSSION THREAD

**Discussion by muriloasouza** [alan-turing-institute/sktime#5554] [Q&A]
I have a timeseries multiclass classification problem (classes from 0 to 11) that i have been using some Sklearn Classifiers ([MLPClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPClassifier.html#sklearn.neural_network.MLPClassifier) and [RandomForestClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html)) and Keras ([Conv1D](https://keras.io/api/layers/convolution_layers/convolution1d/)) to solve it.

Initially, the input to the developed models was the timeseries itself. After that, i tried different combinations of statistical features (mean, variance, standard deviation, etc) as aditional input to my models (to be clear, i am using the timeseries plust the statistical features), wich worked fine and improved my classification performance.

Does these algorithms made specially for times series classification from Sktime (Shapelet or Dictionary based) works only if i provide a  timeseries as input? Or can i continue to provide along with it those statistical features aswell?

**Comment by fkiraly** [comment]
Yes, although, to my knowledge, there is no classifer that would natively support both kinds of features directly - you have to build an ensemble of pipelines.

For a kind of algorithm as you describe, you would first have to build a classifier of either type, then combine them:

* pipeline a series-to-primitives transformer with an skleran classifier, e.g., `SummaryTransformer` with any sklearn classifier. You can also directly use the `SummaryClassifier`
* and pick any sktime classifier
* put both in a `WeightedEnsembleClassifier` or a similar ensemble

This allows you to combine any "sophisticated" time series classifier with one that extracts features and applies an sklearn classifier.

**Reply by muriloasouza** [reply]
Hello @fkiraly , thanks for your answer, that gave me some ideas.

But what do you mean by "no classifier suport both kinds of features"? For example, all those Classifiers i mentioned works. My `x_train` is a numpy array of `N` rows (timeseries) and `27` columns (the first `24` are hourly measurements of the timeseries and the last `3` are mean, variance and standard deviation of the timeseries).
